### PR TITLE
feat: support v2 version for hpa

### PR DESCRIPTION
**Pull request by jiangfucheng**
Fix https://github.com/apache/apisix-helm-chart/issues/463

### CODE REVIEW

**Comment by jiangfucheng** [comment]
@tao12345666333  @tokers  PTAL

**Review by tokers — COMMENTED** [review]
Do we need to add a auto selection feature for users. That is, if the target Kubernetes cluster is appropriate, let's use v2, or let's use `v2beta1`.

**Review by tao12345666333 — CHANGES_REQUESTED** [review]
There is no need to introduce a new template file.
You can directly add conditional judgments to the original template file.

For `autoscaling/v2` and `autoscaling/v2beta1` the changes are not that big enough to warrant a new template file.

**Comment by jiangfucheng** [comment]
> Do we need to add a auto selection feature for users. That is, if the target Kubernetes cluster is appropriate, let's use v2, or let's use `v2beta1`.

Is it feasible to use `.Capabilities.APIVersions.Has "autoscaling/v2"` to check hpa version？

**Comment by jiangfucheng** [comment]
Sorry for delay, I just add a version variable to select the version of HPA, since i find the `.Capabilities.APIVersions.Has` maybe can't check whether k8s cluster supports this version, it just use the version of Helm as check condition. Please review again, @tao12345666333 @tokers .

File: charts/apisix-ingress-controller/README.md [hunk 1]
```diff
@@ -109,6 +109,7 @@ The same for container level, you need to set:
 | autoscaling.maxReplicas | int | `100` |  |
 | autoscaling.minReplicas | int | `1` |  |
 | autoscaling.targetCPUUtilizationPercentage | int | `80` |  |
+| autoscaling.version | string | `"v2"` | HPA version, default v2 |
```
Comment (tokers) [inline charts/apisix-ingress-controller/README.md:112]: Also show the candidate versions here.

Comment (jiangfucheng) [reply charts/apisix-ingress-controller/README.md]: Done.

**Review by tao12345666333 — APPROVED** [review]

**Comment by gkalwig** [comment]
Hi, when you plan release that change? We are unable to update our environment without it, and we really need it.

**Review by Gallardot — APPROVED** [review]

**Pull request merged**